akka-stream/src/main/scala/akka/stream/impl/FlexiRouteImpl.scala

```diff
@@ -31,6 +31,16 @@ private[akka] class FlexiRouteImpl[T, S <: Shape](_settings: ActorFlowMaterializ
   // needed to ensure that at most one element is emitted from onInput
   private val emitted = Array.ofDim[Boolean](outputCount)
 
+  override def preStart(): Unit = {
+    super.preStart()
```

Comment:
if the super throws then we should not attempt the stage preStart IMHO. (i.e. we did not start the stage)